changes to get mpdev verify running

brett060102 opened the pull request:
The following are minimal changes to get mpdev working. The outstanding issue is that enforcer pod does
not get removed and must be deleted in final namespace delete changes:
schema.yaml
  awsbilling. changed to global.aws.
  patch added to verbs list, needed by test deployer
apptest/deployer/schema.yaml
  add missing type: IMAGE
apptest/tester/{Dockerfile,tester.sh}
  simplify down to basic echo test

<!--- /gcbrun -->

rtamalin committed:
WIP Hacking on getting neuvector click2deploy working

Enable billing support by default in the embedded copy of the
neuvector/core under k8s/neuvector/chart/neuvector.

Add README-mpdev.md outlining the steps needed to simulate a Google
marketplace deployment using mpdev.  Also added a template that can
be used to generate the contents for the userinitcfg.yaml secret.

k8s/neuvector/schema.yaml has a first pass at specifying the values
for NeuVector, subject to change.

Added a deployerServiceAccount definition that enables the deployer
to be able to create CRDs. More work needed to add any require RBAC
settings to the deployerServiceAccount to allow it to complete the
deployment, as well as defining any serviceAccounts that are needed
by the running deployment.

Note that the activemq Click to Deploy implementation has been used
as the reference example for helm based application deployment, as
well as the sample-app, which is an envsubst based deployment; still
need to cleanup some remaining ActiveMQ related content and replace
with equivalent for NeuVector.

Currently need to override the definition of C2D_CONTAINER_RELEASE
in the Makefile to specify a hardcoded value as the NeuVector images
don't have the required env setting, C2D_RELEASE, set when they are
being built.

Trim down the apptest test suite somewhat and start to add NeuVector
specific tests.

Add README-chart.md outlining chart maintenance workflow

Define appropriate privs for deployer service account

Also update application description with appropriate instructions
for retrieving the NeuVector webui username and password.

neuvector now neuvector-prime

rtamalin commented on k8s/neuvector/apptest/tester/tester.sh:
```diff
@@ -17,6 +17,6 @@
 set -xeo pipefail
 shopt -s nullglob
 
-for test in /tests/*; do
```
Personally, I would leave the original code in, with the exit 0 above it, to retain the testrunner plumbing example in the code.

rtamalin reviewed (approved):
Looks reasonable...

brett060102 committed:
changes to get mpdev verify running

    The following are minimal changes to get mpdev working.
    The outstanding issue is that enforcer pod does
    not get removed and must be deleted in final namespace delete
    changes:
    schema.yaml
      awsbilling. changed to global.aws.
      patch added to verbs list, needed by test deployer
    apptest/deployer/schema.yaml
      add missing type: IMAGE
    apptest/tester/{Dockerfile,tester.sh}
      simplify down to basic echo test
